cli/src/command/wallet.rs

```diff
@@ -168,16 +204,35 @@ pub async fn restore_command(
     Ok(wallet)
 }
 
-pub async fn set_node_command(wallet: &Wallet, url: String) -> Result<(), Error> {
+pub async fn set_node_command(storage_path: &Path, snapshot_path: &Path, url: String) -> Result<Wallet, Error> {
+    let password = get_password("Stronghold password", !snapshot_path.exists())?;
+    let wallet = unlock_wallet(storage_path, snapshot_path, &password).await?;
     wallet.set_client_options(ClientOptions::new().with_node(&url)?).await?;
 
-    Ok(())
+    Ok(wallet)
 }
 
-pub async fn sync_command(wallet: &Wallet) -> Result<(), Error> {
+pub async fn sync_command(storage_path: &Path, snapshot_path: &Path) -> Result<Wallet, Error> {
+    let password = get_password("Stronghold password", !snapshot_path.exists())?;
+    let wallet = unlock_wallet(storage_path, snapshot_path, &password).await?;
```

Comment:
What's up with all this duplicated code? Looks like it was better before.

Reply:
well then let's not merge it

Reply:
I'm just speaking about this specific change...

Reply:
yeah, talking about the same ... if we undo that we get back to how `new_wallet` fn was implemented before.

Reply:
Ah alright, your answered seemed very passive aggressive :trollface:

Reply:
yeah, I realized it could be picked up this way ... but I tried to reduce repetition as much as possible. I was aware of that. I can look into deduping a bit more as I did with `unlock_wallet`, but honestly ... many commands require similar preconditions, but not identical, otherwise I would have passed them in.

Reply:
I think, as I said in the other comment, that we can dedup by changing the structure of that code rather than separating into another method.

Reply:
But you were talking about a different module, Alex (src/wallet.rs) while this is about (src/command/wallet.rs)

Reply:
Ah, ignore me!

Reply:
all good
@thibault-martinez what exactly is duplicated code? Those are just function calls, not repeated logic code.

Reply:
I found the change worth it since it makes the `match` in the `new_wallet` fn really flat compared to what we had before.

Reply:
I don't think this is repeated code. It was to be explicit but we cant easily just replace the whole thing by `_`

Reply:
I still think that the new impl of `new_wallet` is a lot cleaner because it's doing only one pattern matching on the `cli.command` while the current impl does it 3 times.

Reply:
The difference in processing of these variants exists and we can't do anything about it. You may be removing some matches but you are just moving the difference down the line in the individual functions, adding some more code in each of them that was not previously needed. I guess it's a matter or preference but I personally don't really find it cleaner.

Reply:
yeah, I am moving as much as possible into those functions because that is where I think the logic should be. `new_wallet` for me is a typical control-flow like function that just delegates logic, but doesn't contain a lot of logic itself. It's a preference ... I agree and if people don't like it I revert.